### ISSUE TITLE
Case sensitive: Expected a Workspace ID that matched

### ISSUE THREAD

**Issue by DenisBalan** [terraform-providers/terraform-provider-azurerm#28084]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

The only issue which i found (but its not related is https://github.com/hashicorp/terraform-provider-azurerm/issues/26159 )

tldr

i put the resource id which i got in azure portal (view JSON info)

but its throwing error, because of my value being in lower case fully

ps. 00000000-0000-0000-0000-000000000000 - my masked sub id.

```
│ Error: parsing "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-placeholder/providers/microsoft.operationalinsights/workspaces/law-ddev": parsing segment "staticResourceGroups": parsing the Workspace ID: the segment at position 2 didn't match
│ 
│ Expected a Workspace ID that matched:
│ 
│ > /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.OperationalInsights/workspaces/workspaceValue
│ 
│ However this value was provided:
│ 
│ > /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-placeholder/providers/microsoft.operationalinsights/workspaces/law-ddev
│ 
│ The parsed Resource ID was missing a value for the segment at position 2
│ (which should be the literal value "resourceGroups").

```

strange design to be case sensitive when it could be not.

### Terraform Version

latest

### AzureRM Provider Version

latest

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting.diagnostics (but doesnt matter, applicable for everything i guess)

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "examplestoracc"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier            = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_monitor_diagnostic_setting" "example" {
  name                       = "example-diagnostic-setting"
  target_resource_id         = azurerm_storage_account.example.id
  log_analytics_workspace_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-placeholder/providers/microsoft.operationalinsights/workspaces/law-ddev" # existing one!

  enabled_log {
    category = "StorageWrite"
    enabled  = true
  }

  enabled_log {
    category = "StorageRead"
    enabled  = true
  }

  retention_policy {
    enabled = true
    days    = 30
  }
}
```

### Debug Output/Panic Output

```shell
see above
```

### Expected Behaviour

to handle even case insensitive resource ids (review all, not only log analytics resource pointers)

### Actual Behaviour

case sensitive check in place.

**Comment by teowa** [comment]
Hi @DenisBalan, this is by design that the `log_analytics_workspace_id` field only accepts case sensitive ID for now. Please workaround this by using a data source:
```hcl
data "azurerm_log_analytics_workspace" "example" {
  name                = "acctest-01"
  resource_group_name = "acctest"
}

resource "azurerm_monitor_diagnostic_setting" "example" {
  name                       = "example-diagnostic-setting"
  target_resource_id         = azurerm_storage_account.example.id
  log_analytics_workspace_id  = data.azurerm_log_analytics_workspace.example.id
...
...
}
```

Thanks.

**Comment by m-soltani** [comment]
I am having the same issue by importing existing resource into state file.
The case sensitivity is an extra load, since I have to touch change a lot of
commands to make match the case sensitivity.
